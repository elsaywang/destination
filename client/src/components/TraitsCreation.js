import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Add from '@react/react-spectrum/Icon/Add';
import Button from '@react/react-spectrum/Button';
import Link from '@react/react-spectrum/Link';
import styles from './TraitsCreation.css';

class TraitsCreation extends Component {
    singleRowTraitsCreation = () => {
        const { traitsCreationLabelText } = this.props;
        return (
            <Link href="#">
                <div className={styles.singleCreation}>
                    <Add size="S" />
                    <div className={styles.singleCreationlinkText}>{traitsCreationLabelText}</div>
                </div>
            </Link>
        );
    };

    multiRowsTraitsCreation = () => {
        const { handleTraitsCreation, selectedSignals, traitsCreationLabelText } = this.props;

        const { selectionMessage, warning } = selectedSignals;

        return (
            <div className={styles.multiCreation}>
                <span className={styles.message}>{selectionMessage}</span>
                <Button
                    label={traitsCreationLabelText}
                    icon={<Add />}
                    onClick={handleTraitsCreation}
                    variant="action"
                    disabled={warning}
                />
            </div>
        );
    };

    render() {
        const { multiCreation, handleTraitsCreation } = this.props;
        return multiCreation ? this.multiRowsTraitsCreation() : this.singleRowTraitsCreation();
    }
}

TraitsCreation.propTypes = {
    multiCreation: PropTypes.bool,
    selectedSignals: PropTypes.object,
    traitsCreationLabelText: PropTypes.string,
    handleTraitsCreation: PropTypes.func,
};

export default TraitsCreation;
