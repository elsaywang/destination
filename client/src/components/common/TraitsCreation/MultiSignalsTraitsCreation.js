import React from 'react';
import PropTypes from 'prop-types';
import styles from './TraitsCreation.css';
import Button from '@react/react-spectrum/Button';
import Add from '@react/react-spectrum/Icon/Add';

const MultiSignalsTraitsCreation = ({
    selectedSignals,
    storeSessionAndNavigateToTraits,
    isMaxSignalSelectionsReached,
}) => {
    const {
        selectionMessage,
        hasSignalSelectionsTypeWarning,
        hasOnboardedSignalSelectionsWarning,
    } = selectedSignals;

    const isTraitCreationDisabled =
        hasSignalSelectionsTypeWarning ||
        isMaxSignalSelectionsReached ||
        hasOnboardedSignalSelectionsWarning;

    return (
        <div className={styles.multiCreation}>
            <span className={styles.message}>{selectionMessage}</span>
            <Button
                onClick={storeSessionAndNavigateToTraits}
                label="Create Trait From Multiple Signals"
                icon={<Add />}
                variant="action"
                disabled={isTraitCreationDisabled}
                data-test="multi-signals-trait-creation"
            />
        </div>
    );
};

MultiSignalsTraitsCreation.propTypes = {
    selectedSignals: PropTypes.shape({
        selectionMessage: PropTypes.string,
        hasSignalSelectionsTypeWarning: PropTypes.bool,
        hasOnboardedSignalSelectionsWarning: PropTypes.bool,
        selectedRowIndexes: PropTypes.array,
    }),
    storeSessionAndNavigateToTraits: PropTypes.func.isRequired,
    isMaxSignalSelectionsReached: PropTypes.bool,
};

export default MultiSignalsTraitsCreation;
