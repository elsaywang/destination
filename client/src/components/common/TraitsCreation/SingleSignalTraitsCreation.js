import React from 'react';
import PropTypes from 'prop-types';
import styles from './TraitsCreation.css';
import Add from '@react/react-spectrum/Icon/Add';
import Link from '@react/react-spectrum/Link';

const SingleSignalTraitsCreation = ({ traitsCreationLabelText }) => {
    return (
        <Link href="#">
            <div className={styles.singleCreation}>
                <Add size="S" />
                <div className={styles.singleCreationlinkText}>{traitsCreationLabelText}</div>
            </div>
        </Link>
    );
};

SingleSignalTraitsCreation.propTypes = {
    traitsCreationLabelText: PropTypes.string,
};

export default SingleSignalTraitsCreation;
