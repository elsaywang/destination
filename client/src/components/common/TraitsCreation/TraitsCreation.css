:local(.singleCreationlinkText) {
    display: inline-flex;
    font-size: 0.875rem;
    padding-left: 0.15em;
    vertical-align: text-bottom;
}

:local(.singleCreation) {
    cursor: pointer;
    margin-left: -1.05em;
    white-space: nowrap;
}

:local(.multiCreation) {
    display: flex;
    justify-content: flex-end;
}

:local(.message) {
    display: flex;
    padding: 0.5em;
    font-weight: 800;
    white-space: nowrap;
}
