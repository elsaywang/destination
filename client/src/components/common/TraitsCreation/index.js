import TraitsCreation from './TraitsCreation';

export default TraitsCreation;
