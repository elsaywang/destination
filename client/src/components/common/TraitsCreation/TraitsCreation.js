import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MultiSignalsTraitsCreation from './MultiSignalsTraitsCreation';
import SingleSignalTraitsCreation from './SingleSignalTraitsCreation';

class TraitsCreation extends Component {
    render() {
        const {
            multiCreation,
            traitsCreationLabelText,
            selectedSignals,
            handleTraitsCreation,
        } = this.props;
        return multiCreation ? (
            <MultiSignalsTraitsCreation
                {...{ handleTraitsCreation, selectedSignals, traitsCreationLabelText }}
            />
        ) : (
            <SingleSignalTraitsCreation {...{ traitsCreationLabelText }} />
        );
    }
}

TraitsCreation.propTypes = {
    multiCreation: PropTypes.bool,
    selectedSignals: PropTypes.shape({
        selectionMessage: PropTypes.string,
        warning: PropTypes.bool,
        records: PropTypes.array,
    }),
    traitsCreationLabelText: PropTypes.string,
    handleTraitsCreation: PropTypes.func,
};

export default TraitsCreation;
