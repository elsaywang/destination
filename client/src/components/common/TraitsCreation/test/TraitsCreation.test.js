import React from 'react';
import { shallow } from 'enzyme';
import TraitsCreation from '../TraitsCreation';
import Add from '@react/react-spectrum/Icon/Add';
import Link from '@react/react-spectrum/Link';
import MultiSignalsTraitsCreation from '../MultiSignalsTraitsCreation';
import SingleSignalTraitsCreation from '../SingleSignalTraitsCreation';

describe('<TraitsCreation/> component', () => {
    describe('rendering when it is used in Single-Signal Traits Creation', () => {
        const mockFn = jest.fn();
        const props = {
            multiCreation: false,
            selectedSignals: { selectionMessage: '', warning: false },
            traitsCreationLabelText: 'Create Onboarded Trait',
            handleTraitsCreation: mockFn,
        };

        const wrapper = shallow(<TraitsCreation {...props} />);
        const creationWrapper = wrapper.instance();

        it('matches the snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders with prop `multiCreation` is false', () => {
            expect(creationWrapper.props.multiCreation).toEqual(false);
        });

        it('renders <SingleSignalTraitsCreation/> component', () => {
            expect(wrapper.find(SingleSignalTraitsCreation).exists()).toBe(true);
        });

        it('<SingleSignalTraitsCreation/> has `traitsCreationLabelText` props with correct text', () => {
            expect(
                wrapper.find(SingleSignalTraitsCreation).props().traitsCreationLabelText,
            ).toEqual(props.traitsCreationLabelText);
        });
    });

    describe('rendering when it is used in Multi-Signals Traits Creation', () => {
        const mockFn = jest.fn();
        const props = {
            multiCreation: true,
            selectedSignals: { selectionMessage: '', warning: false },
            traitsCreationLabelText: 'Create Trait From Multi Signals',
            handleTraitsCreation: mockFn,
        };

        const wrapper = shallow(<TraitsCreation {...props} />);
        const creationWrapper = wrapper.instance();

        it('matches the snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders with prop `multiCreation` is true', () => {
            expect(creationWrapper.props.multiCreation).toEqual(true);
        });

        it('renders <MultiSignalsTraitsCreation/> component', () => {
            expect(wrapper.find(MultiSignalsTraitsCreation).exists()).toBe(true);
        });

        it('<MultiSignalsTraitsCreation/> has `traitsCreationLabelText` props with correct text', () => {
            expect(
                wrapper.find(MultiSignalsTraitsCreation).props().traitsCreationLabelText,
            ).toEqual(props.traitsCreationLabelText);
        });
    });
});
