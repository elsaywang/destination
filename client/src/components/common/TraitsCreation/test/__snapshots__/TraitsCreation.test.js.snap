// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TraitsCreation/> component rendering when it is used in Multi-Signals Traits Creation matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TraitsCreation
    handleTraitsCreation={[MockFunction]}
    multiCreation={true}
    selectedSignals={
      Object {
        "selectionMessage": "",
        "warning": false,
      }
    }
    traitsCreationLabelText="Create Trait From Multi Signals"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "handleTraitsCreation": [MockFunction],
      "selectedSignals": Object {
        "selectionMessage": "",
        "warning": false,
      },
      "traitsCreationLabelText": "Create Trait From Multi Signals",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "handleTraitsCreation": [MockFunction],
        "selectedSignals": Object {
          "selectionMessage": "",
          "warning": false,
        },
        "traitsCreationLabelText": "Create Trait From Multi Signals",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<TraitsCreation/> component rendering when it is used in Single-Signal Traits Creation matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TraitsCreation
    handleTraitsCreation={[MockFunction]}
    multiCreation={false}
    selectedSignals={
      Object {
        "selectionMessage": "",
        "warning": false,
      }
    }
    traitsCreationLabelText="Create Onboarded Trait"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "traitsCreationLabelText": "Create Onboarded Trait",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "traitsCreationLabelText": "Create Onboarded Trait",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
